jupyterfs/manager/fsspec.py

```diff
@@ -250,8 +290,17 @@ def get(self, path, content=True, type=None, format=None):
 
     def _save_directory(self, path, model):
         """create a directory"""
-        if not self._fs.exists(path):
-            self._fs.mkdir(path)
+        if not self.allow_hidden and self.is_hidden(path):
+            raise web.HTTPError(400, f"Cannot create directory {path!r}")
+
+        if not self.exists(path):
+            # TODO better carveouts
+            if self._fs.__class__.__name__.startswith("S3"):
+                # need to make a file temporarily
+                # use the convention of a hidden file
+                self._fs.touch(f"{path}/.s3fskeep")
```

Comment:
folders aren't real in s3, i adopted the convention many s3 backends do which is to force the creation of a "directory" by creating a file. e.g. on backblaze its called `.bzEmpty`.